### ISSUE TITLE
lazysizes.client.js running on server

### ISSUE THREAD

**Issue by CHEWX** [nuxt/nuxt#29056]
### Environment

```
------------------------------
- Operating System: Darwin
- Node Version:     v18.18.2
- Nuxt Version:     3.13.0
- CLI Version:      3.13.1
- Nitro Version:    2.9.7
- Package Manager:  unknown
- Builder:          -
- User Config:      ssr, nitro, compatibilityDate, devtools, app, modules, runtimeConfig, plugins, hooks, sitemap
- Runtime Modules:  nuxt-fathom@0.0.2, @nuxtjs/sitemap@6.0.1
- Build Modules:    -
------------------------------
```

### Reproduction

```
import lazySizes from 'lazysizes'
import 'lazysizes/plugins/parent-fit/ls.parent-fit'

export default defineNuxtPlugin(() => {
    lazySizes.init()
})
```
`plugins/lazysizes.client.js`

I get this error.

 - rendered on server: class="c-product__image lazyloading"

### Describe the bug

As above, lazyloading shouldn't be applied on server as the plugin shouldn't be running...

**Comment by simonhrogers** [comment]
Here is one:

https://layout-transition-no-worky-nuxt3.netlify.app/

https://github.com/simonhrogers/transition-no-worky

My workaround for this, has been to wrap the import in a client check, but this appears to negatively affect load times in pagespeed scores:

```
  if (import.meta.client) {
    import('lazysizes')
    import('lazysizes/plugins/parent-fit/ls.parent-fit')
  }
  ```

**Comment by CHEWX** [comment]
Seems counter intuitive considering the suffix should be sufficient enough as a "client check".

**Comment by simonhrogers** [comment]
Oh absolutely, the plugin should only be loaded on the client as stated at the top of the Nuxt docs:

`You can use .server or .client suffix in the file name to load a plugin only on the server or client side.`

I’m just sharing a quick workaround as I don‘t know how quickly the Nuxt team is currently able to respond to issues like this!  It feels like a fundamental part of the framework not working as advertised to me, as does my open issue on layout transitions, but they are a small team doing a heroic amount and I imagine Nuxt 4 is taking a lot of headspace.

**Comment by danielroe** [comment]
In your reproduction, the plugin is not running on the server, nor are the classes `c-product__image lazyloading` rendered on either page.

Note that you do not need to add the plugin in the `plugins` array in nuxt config - they are scanned automatically.

Would you explain more, or provide a working reproduction? 🙏

**Comment by CHEWX** [comment]
Good to know about the plugins scanning.

>nor are the classes c-product__image lazyloading rendered on either page.

https://github.com/simonhrogers/transition-no-worky/blob/main/pages/index.vue#L93

See the above, ignore the `.c-product_image`, that's from my setup. It's the `.lazyloading` class were looking at.

**Comment by CHEWX** [comment]
It's actually nothing to do with the `.init()` either, it's running on the server just via the `import`

```
import lazySizes from 'lazysizes'
// import 'lazysizes/plugins/parent-fit/ls.parent-fit'

export default defineNuxtPlugin(() => {
//     lazySizes.init()
})
```

**Comment by danielroe** [comment]
No, the `lazyloading` class isn't added either in the server-rendered HTML.

I've added a console log and fixed the incorrect HTML. You can see it's working fine: https://stackblitz.com/edit/github-xdvnbl?file=plugins/lazysizes.client.js,pages/about.vue,pages/index.vue

**Comment by CHEWX** [comment]
The problem, is the OP is not linked to the hydration of the kindly provided reproduction.

Here is my error. Perhaps, it's nothing to do with lazysizes, it's just the "rendered on server" is giving us a red herring? As `lazyloading` or `lazyloaded` should never happen on server.

<img width="867" alt="Screenshot 2024-09-19 at 13 18 57" src="https://github.com/user-attachments/assets/76258bae-e09c-46ed-b56d-c88ff7e460a2">

**Comment by danielroe** [comment]
The error is almost certainly related to the lazy loading code running before Nuxt is hydrated.

That's why I suggested using `onNuxtReady`.

But if the code runs _on import_ and not when calling `init()` then you may have to use a dynamic import as you suggested.

**Comment by simonhrogers** [comment]
@danielroe Sorry my bad – you’re right – it works perfectly in my reproduction once the nuxt.config plugins array was removed, mistake on my part as I thought it was still needed (I’ve been a Nuxt user since V1). 

If you do have time to look into https://github.com/nuxt/nuxt/issues/28988, which is what I originally prepared the reproduction for, I would really appreciate it as it would save me doing some fairly massive/hacky workarounds to get it to appear as if there were different layouts. If you have time this week or next, I would love to buy you a 1:1 bouldering coaching session with Louis Parkinson should you ever be down in London :-)

**Comment by danielroe** [comment]
you know the way to my heart

**Comment by simonhrogers** [comment]
indeed – there is only bouldering 

...non-London based coach also offered, but I have only been bouldering a couple of years and had a lovely time with Louis and a couple of friends in May which has seriously helped all three of us get the most out of our sessions :-) Mike Mullins in Scotland I hear is also great
